### ISSUE TITLE
STWO Client Side Proving

### ISSUE THREAD

**Issue by drspacemn** [keep-starknet-strange/backlog#55]
# Client-side proving unlocks on Starknet
TL;DR: should we start hyping and pinning down customers?

Stwo + a Stwo verifier contract on Starknet will unlock :zap: lightning-fast e2e flows of prove locally → verify on Starknet. 

Two example use cases:
- Private applications with seamless UX
- Richer validate that can verify STARK proofs (increasing its capacity to essentially unbounded compute; [see more details here](https://docs.google.com/document/d/1zex0gkElYzUoTaq9stx3SP_b02BJAWnjuchB9ZgVW9I/edit#heading=h.avfhc8lptq8s))

Especially for private applications, perhaps it’s something we should start selling/hyping/encouraging/discussing? Perhaps this can be a game changer for Starknet if we market it correctly?

## Ideas: 
- atomic swap offchain, proved onchain
- zkml, zklogin

**Comment by drspacemn** [comment]
https://starkwareindustries.slack.com/archives/C047BCNSXT3/p1728839246675679
